R/workhorse.R

```diff
@@ -34,9 +31,8 @@ set_path <- function(x) {
   if (!is.null(x$path)) {
     return(x)
   }
-  ## find the path that matches the given method and resource
-  x$path <- unique(.drive$params$path[.drive$params$method == x$method &
-                                        .drive$params$resource == x$resource])
+  ## find the path that matches the given endpoint
+  x$path <- unique(.drive$params$path[.drive$params$endpoint == x$endpoint])
   return(x)
```

Comment:
I'm not sure of the exact structure of `.drive$params` but could you take an approach like I've done here:

https://github.com/tidyverse/googlesheets2/blob/8caf705a92b7c0638783ce478eb0449a3a92c1de/R/gs_build_request.R#L47

where you store the row or component corresponding to this endpoint as, say, `ept`. Then get to make much simpler calls to extract bits? The `unique()` catches my eye. Are we actually at risk of having things ... of length greater than 1?